sklearn/preprocessing/data.py

```diff
@@ -1917,3 +1929,129 @@ def transform(self, X):
+    feature_indices: "all" or array of indices or mask, default None
+        Specify what features are treated are to be transformed.
+
+        - 'all' (default): All features are to be transformed.
+        - array of indices: Array of feature indices to be transformed..
+        - mask: Array of length n_features and with dtype=bool.
+    Notes
+    -----
+    The Box-Cox transform is given by::
+
+        y = (x**lmbda - 1) / lmbda,  for lmbda > 0
+            log(x),                  for lmbda = 0
+
+    `boxcox` requires the input data to be positive.
+
+    This estimator is stateless (besides constructor parameters), the
+    fit method does nothing but is useful when used in a pipeline.
+
+    See also
+    --------
+    boxcox_transform: Equivalent function without the object oriented API.
+    """
+
+    def __init__(self, feature_indices=None, copy=False):
+        self.feature_indices = feature_indices
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to fit by apply boxcox transform,
+            to each of the features and learn the lambda.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        if any(np.any(X <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied on positive data")
+        n_features = X.shape[1]
+        outputs = Parallel(n_jobs=-1)(delayed(_boxcox)(X[:, i], None, True) for i in range(n_features))
+        output = np.concatenate([o[0][..., np.newaxis] for o in outputs], axis=1)
+        self.lambdas = np.array([o[1] for o in outputs])
+        return self
+
+    def transform(self, X, y=None, feature_indices=None, copy=None):
+        """Scale each non zero row of X to unit norm
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to apply boxcox transform, to each of the
+            features given in the ``feature_indices`` attribute.
+        """
+        copy = copy if copy is not None else self.copy
+        selected = feature_indices if feature_indices is not None else "all"
+        n_features = X.shape[1]
+        if any(np.any(X <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied on positive data")
+        if isinstance(feature_indices, np.ndarray):
+            if feature_indices.dtype == np.bool:
+                self.indices = np.where(feature_indices)[0]
+            else:
+                self.indices = feature_indices            
+        else:
+                self.indices = np.arange(n_features)
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        X_tr = _transform_selected(X, self._transform, selected, copy, order=True)
+        return X_tr
+
+    def _transform(self, X):
+        maxlogs = [self.lambdas[i] for i in self.indices]
```

Comment:
In _boxcox, there is a input parameter lambda which is used for transform and also a return parameter when using for fit. I was doing this to disambiguate the use case scenario.